### PR TITLE
Do not generate keys that are too long for LMDB

### PR DESCRIPTION
**Pull request by Kerollmops**
This PR fixes https://github.com/meilisearch/meilisearch/issues/2338 by making sure that we do not generate keys that are too long for LMDB especially when we are creating our prefix and proximity pairs keys.

### CODE REVIEW

File: milli/src/update/index_documents/mod.rs [hunk 1]
```diff
@@ -1651,4 +1652,60 @@ mod tests {
...
...nkey for a moment before roaring very \nloudly.\r\nhttps://bug.example.com/meilisearch/milli.saml2?ROLE=Programmer-1337&SAMLRequest=Cy1ytcZT1Po%2L2IY2y9Unru8rgnW4qWfPiI0EpT7P8xjJV8PeQikRL%2E8D9A4pj9tmbymbQCQwGmGjPMK7qwXFPX4DH52JO2b7n6TXjuR7zkIFuYdzdY2rwRNBPgCL7ihclEm9zyIjKZQ%2JTqiwfXxWjnI0KEYQYHdwd6Q%2Fx%28BDLNsvmL54CCY2F4RWeRs4eqWfn%2EHqxlhreFzax4AiQ2tgOtV5thOaaWqrhZD%2Py70nuyZWNTKwciGI43AoHg6PThANsQ5rAY5amzN%2ufbs1swETUXlLZuOut5YGpYPZfY6STJWNp4QYSUOUXBZpdElYsH7UHZ7VhJycgyt%28aTK0GW6GbKne2tJM0hgSczOqndg6RFa9WsnSBi4zMcaEfYur4WlSsHDYInF9ROousKqVMZ6H8%2gbUissaLh1eXRGo8KEJbyEHbhVVKGD%28kx4cfKjx9fT3pkeDTdvDrVn25jIzi9wHyt9l1lWc8ICnCvXCVUPP%2BjBG4wILR29gMV9Ux2QOieQm2%2Fycybhr8sBGCl30mHC7blvWt%2T3mrCHQoS3VK49PZNPqBZO9C7vOjOWoszNkJx4QckWV%2FZFvbpzUUkiBiehr9F%2FvQSxz9lzv68GwbTu9fr638p%2FQM%3D&RelayState=https%3A%2F%example.bug.com%2Fde&SigAlg=http%3A%2F%2Fwww.w3.org%2F2000%2F09%2Fxmldsig%23rsa-sha1&Signature=AZFpkhFFII7PodiewTovaGnLQKUVZp0qOCCcBIUkJ6P5by3lE3Lldj9pKaFu4wz4j%2B015HEhDvF0LlAmwwES85vdGh%2FpD%2cIQPRUEjdCbQkQDd3dy1mMXbpXxSe4QYcv9Ni7tqNTQxekpO1gE7rtg6zC66EU55uM9aj9abGQ034Vly%2F6IJ08bvAq%2B%2FB9KruLstuiNWnlXTfNGsOxGLK7%2BXr94LTkat8m%2FMan6Qr95%2KeR5TmmqaQIE4N9H6o4TopT7mXr5CF2Z3\n";
```
Comment (ManyTheFish) [inline milli/src/update/index_documents/mod.rs:1662]: Could we make this smaller in order to test only what we want to test?

**Review by ManyTheFish — CHANGES_REQUESTED** [review]
You seem to want to `unwrap` each key insertion in the database, there are some in [typed_chunk.rs](https://github.com/meilisearch/milli/blob/312515dd6b676dcc33e1865e2521370b29c9872b/milli/src/update/index_documents/typed_chunk.rs#L260-L341).
Is it relevant for you to unwrap them too?

**Comment by MarinPostma** [comment]
Why weren't we able to get the error ? And why is unwraping here a reasonable fix?

File: milli/src/update/word_prefix_docids.rs [hunk 2]
```diff
@@ -124,7 +125,9 @@ fn write_prefixes_in_sorter(
 ) -> Result<()> {
     for (key, data_slices) in prefixes.drain() {
         for data in data_slices {
-            sorter.insert(&key, data)?;
+            if valid_lmdb_key(&key) {
+                sorter.insert(&key, data)?;
+            }
```
Comment (ManyTheFish) [inline milli/src/update/word_prefix_docids.rs:130]: @MarinPostma, I understand that the fix is here. The unwraps later are considered unreachable.

Comment (Kerollmops) [reply milli/src/update/word_prefix_docids.rs:130]: I removed the unwraps and replaced them back by the `?` operator. That was a mistake.

File: milli/src/update/index_documents/mod.rs (hunk 1 above)
Comment (Kerollmops) [reply milli/src/update/index_documents/mod.rs]: Reduced!

**Comment by Kerollmops** [comment]
Thank you both for your reviews,

The unwraps were a mistake, sorry, I didn't take the time to review before sending the PR. It will not happen again!
I reduced the test data to a smaller text, thank you.

**Review by ManyTheFish — APPROVED** [review]

**Review by Kerollmops — COMMENTED** [review]
bors merge

**Comment by Kerollmops** [comment]
Hey @dvic,

I read that too but it, indeed, impacts the performances and I prefer that we keep most of the settings by default, it is easier to debug and the default maximum key size is already set to a big enough value (512bytes), which mean a pair of words of length ~210 bytes can fit.

Thank you very much for your interest in the project!

**Pull request merged**